File: samples/java-azurefunction/src/main/java/com/function/CreateNewOrder.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package main.java.com.function;
import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;
import com.microsoft.azure.functions.dapr.annotation.DaprServiceInvocationTrigger;
import com.microsoft.azure.functions.dapr.annotation.DaprStateOutput;
import com.microsoft.azure.functions.OutputBinding;

import java.util.Optional;

/**
 * Azure Functions with Dapr service invocation trigger.
 */
public class CreateNewOrder {
    /**
     * This function gets invoked by dapr runtime:
     * dapr invoke --app-id functionapp --method CreateNewOrder --data '{"value": { "orderId": "41" } }'
     */
    @FunctionName("CreateNewOrder")
    public String run(
            @DaprServiceInvocationTrigger(
                methodName = "CreateNewOrder") 
            String payload,
            @DaprStateOutput(
                stateStore = "%StateStoreName%",
                key = "order")
            OutputBinding<String> product,
            final ExecutionContext context) {
        context.getLogger().info("'Java function processed a CreateNewOrder request from the Dapr Runtime.'");
        product.setValue(payload);

        return payload;
    }
}

File: samples/javascript-v4-azurefunction/src/functions/createNewOrder.js
const { app, output, trigger } = require('@azure/functions');

const daprStateOuput = output.generic({
    type: "daprState",
    stateStore: "%StateStoreName%",
    direction: "out",
    name: "order",
    key: "order"
});

app.generic('CreateNewOrder', {
    trigger: trigger.generic({
        type: 'daprServiceInvocationTrigger',
        name: "payload"
    }),
    return: daprStateOuput,
    handler: async (request, context) => {
        context.log("Node function processed a CreateNewOrder request from the Dapr Runtime.");
        context.log(context.triggerMetadata.payload.data)

       return context.triggerMetadata.payload.data;
    }
});